test/direct.jl

```diff
@@ -180,3 +180,55 @@ end
     self_traces = filter(entry -> string(entry.file) == @__FILE__, stacktraces[1])
     @test length(self_traces) == 3
 end
+
+if MKL_jll.is_available() && Sys.WORD_SIZE == 64
+    mkl_libdir = dirname(MKL_jll.libmkl_rt_path)
+    mkl_libs = readdir(mkl_libdir)
+    find_mkl_lib(name) = joinpath(mkl_libdir, mkl_libs[findfirst(l -> occursin(name, l), mkl_libs)])
+
+    # Load MKL_jll in explicit ILP64 mode.  We need to manually load the `core` library,
+    # and the appropriate threading library.  We choose intel threading here.
+    dlopen(find_mkl_lib("mkl_core"), RTLD_LAZY | RTLD_GLOBAL)
+    dlopen(find_mkl_lib("mkl_intel_thread"), RTLD_LAZY | RTLD_GLOBAL)
+
+    # Since MKL v2022, we can manually load an ILP64-only library
+    @testset "MKL v2022 ILP64 loading" begin
+        # Load the ILP64 interface library.  Remember, you must load the `core`
+        # and a `threading` library first, with `RTLD_LAZY` for this to work!
```

Comment:
Oh wow, you need to find the right incantation ti make this work: https://github.com/JuliaLinearAlgebra/libblastrampoline/issues/52#issuecomment-1003614836